source/draw/gpu/GpuCanvasYuv420Semiplanar.ooc

```diff
@@ -76,9 +76,10 @@ GpuCanvasYuv420Semiplanar: class extends GpuCanvas {
 		uvLines free()
 	}
 	drawPoints: override func (pointList: VectorList<FloatPoint2D>) { this _target y canvas drawPoints(pointList) }
-	fill: override func {
-		this _target y canvas fill()
-		this _target uv canvas fill()
+	fill: override func ~color (color: ColorRgba) {
+		yuv := color toYuv()
+		this _target y canvas fill(ColorRgba new(yuv y, 0, 0, 255))
+		this _target uv canvas fill(ColorRgba new(yuv u, yuv v, 0, 255))
```

Comment:
The hard thing is that monochrome and UV does not always belong to a YUV image. UV has become a default format for any packed two channel image and can represent hue and saturation from the HSV color system.